lib/src/whisper_manager.py

```diff
@@ -87,7 +118,134 @@ def _detect_gpu_backend(self) -> str:
+
+    def _transcribe_rest(self, audio_data: np.ndarray, sample_rate: int = 16000) -> str:
+        """
+        Transcribe audio using remote REST API endpoint
+
+        Args:
+            audio_data: NumPy array of audio samples (float32)
+            sample_rate: Sample rate of the audio data
+
+        Returns:
+            Transcribed text string
+        """
+        try:
+            import requests
+
+            # Get REST endpoint configuration
+            endpoint_url = self.config.get_setting('rest_endpoint_url')
+            api_key = self.config.get_setting('rest_api_key')
+            timeout = self.config.get_setting('rest_timeout', 30)
+
+            if not endpoint_url:
+                raise ValueError('REST endpoint URL not configured')
+
+            print(f'[TRANSCRIBE] Using remote REST API: {endpoint_url}', flush=True)
+
+            # Convert audio to WAV format
+            wav_bytes = self._numpy_to_wav_bytes(audio_data, sample_rate)
+            print(
+                f'[REST] Converted audio to WAV format ({len(wav_bytes)} bytes)',
+                flush=True,
+            )
+
+            # Prepare the request
+            files = {'file': ('audio.wav', wav_bytes, 'audio/wav')}
+
+            headers = {'Accept': 'application/json'}
+            if api_key:
+                headers['Authorization'] = f'Bearer {api_key}'
+
+            # Add language parameter if configured
+            data = {}
+            language = self.config.get_setting('language', None)
+            if language:
+                data['language'] = language
+
+            # Send the request
+            print(f'[REST] Sending request to {endpoint_url}...', flush=True)
+            response = requests.post(endpoint_url, files=files, data=data, headers=headers, timeout=timeout)
+
+            # Check for HTTP errors
+            if response.status_code != 200:
+                error_msg = f'REST API returned status {response.status_code}'
+                try:
+                    error_detail = response.json()
+                    error_msg += f': {error_detail}'
+                except:
+                    error_msg += f': {response.text[:200]}'
+                print(f'ERROR: {error_msg}')
+                return ''
+
+            # Parse the response
+            result = response.json()
+
+            # Try common response formats
+            transcription = ''
+            if 'text' in result:
+                transcription = result['text']
+            elif 'transcription' in result:
+                transcription = result['transcription']
+            elif 'result' in result:
+                transcription = result['result']
+            else:
+                print(f'ERROR: Unexpected response format: {result}')
+                return ''
+
+            print(
+                f'[REST] Transcription received ({len(transcription)} chars)',
+                flush=True,
+            )
+            return transcription.strip()
```

Comment:
Our method gets a little bit of wobble here. We're not forming an actual contract against an API, so we need to be flexible so that we can support arbitrary backends within reason. And this is a start to that, but it feels flaky.

Something to think about. 

But it's okay for now.